Audacia.Log.AspNetCore/HttpDependencyBodyCaptureTelemetryInitializer.cs

```diff
@@ -0,0 +1,181 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Extensions.Options;
+
+namespace Audacia.Log.AspNetCore;
+
+/// <summary>
+/// Enriches http dependency telemetry with request and response bodies.
+/// </summary>
+public class HttpDependencyBodyCaptureTelemetryInitializer : ITelemetryInitializer
+{
+    /// <summary>
+    /// Gets the custom property name for the request body.
+    /// </summary>
+    public const string RequestBodyCustomProperty = "RequestBody";
+
+    /// <summary>
+    /// Gets the custom property name for the response body.
+    /// </summary>
+    public const string ResponseBodyCustomProperty = "ResponseBody";
+
+    /// <summary>
+    /// Gets or sets the max depth for deconstructing objects in the request body.
+    /// </summary>
+    public int MaxDepth { get; set; } = 32;
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the logging of all data in the request body is disabled.
+    /// </summary>
+    public bool DisableBodyContent { get; set; }
+
+    /// <summary>Gets the names of arguments to exclude from the logs.</summary>
+    public ICollection<string> ExcludeProperties { get; } =
+        new List<string>
+        {
+            "username",
+            "password",
+            "access_token",
+            "email",
+            "token",
+            "bearer",
+            "name",
+            "firstname",
+            "lastname",
+            "phonenumber",
+            "dateofbirth"
+        };
+
+    /// <summary>
+    /// Creates telemetry initialiser for enriching dependency telemetry with request and response bodies.
+    /// </summary>
+    /// <param name="options"></param>
+    public HttpDependencyBodyCaptureTelemetryInitializer(IOptions<LogActionFilterConfig> options)
+    {
+        Configure(options?.Value);
+    }
+
+    /// <summary>
+    /// Applies configuration to the log filter if provided.
+    /// </summary>
+    /// <param name="config">global or action config.</param>
+    private void Configure(LogActionFilterConfig? config)
+    {
+        if (config == null)
+        {
+            return;
+        }
+
+        DisableBodyContent = config.DisableBodyContent;
```

Comment:
Don't think this is used?

Do we perhaps want config for `DisableResponseBody` and `DisableRequestBody` so you only have to log one or the other? Or two separate initializers, and you add one or both?

Reply:
Added more granual control.